testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestReadUniontype.java

```diff
@@ -167,6 +167,39 @@ public void testUnionTypeSchemaEvolution(String storageFormat)
         }
     }
 
+    @Test(groups = SMOKE)
+    public void testReadOrcUniontypeWithCheckpoint()
```

Comment:
Thanks @baohe-zhang , great to see this being reproduced. 

```
2022-11-06T15:50:09.2852993Z tests               | 2022-11-06 21:35:09 INFO: Test io.trino.tests.product.hive.TestReadUniontype.testReadOrcUniontypeWithCheckpoint took 42.67s
2022-11-06T15:50:09.9087769Z tests               | 2022-11-06 21:35:09 INFO: SUCCESS     /    io.trino.tests.product.hive.TestReadUniontype.testReadOrcUniontypeWithCheckpoint (Groups: smoke) took 43.3 seconds
2022-11-06T15:50:11.0978592Z tests               | 2022-11-06 21:35:11 INFO: [141 of 561] io.trino.tests.product.hive.TestReadUniontype.testReadUniontype [ORC] (Groups: smoke)
```

The test takes ~43s on the CI. I'm bit on the fence whether it "deserves" this kind of time - given it was an omission originally. This is ~1% of overall module time - less than many other tests. So I'm leaning towards including it. cc @findepi @electrum